header Header
int numErrors
quadError errorMessages[]